plugins/inventory/windows_azure.py

```diff
@@ -82,15 +90,39 @@ def __init__(self):
 
         if self.args.list_images:
             data_to_print = self.json_format_dict(self.get_images(), True)
-        elif self.args.list:
+        elif self.args.list or self.args.host:
             # Display list of nodes for inventory
             if len(self.inventory) == 0:
-                data_to_print = self.get_inventory_from_cache()
+                data = json.loads(self.get_inventory_from_cache())
             else:
-                data_to_print = self.json_format_dict(self.inventory, True)
+                data = self.inventory
 
+            if self.args.host:
+                data_to_print = self.get_host(self.args.host)
+            else:
+                # Add the `['_meta']['hostvars']` information.
+                hostvars = {}
+                if len(data) > 0:
+                    for host in set(reduce(lambda x, y: x + y,
+                                           [i for i in data.values()])):
```

Comment:
Not sure about this section.
* What's in data.values()?  If it's numbers then you could do ```sum(data.values())``` instead.  If it's strings, then you can do ```"".join(data.values())```.
* I think that reduce returns a scalar.  So there's no reason to turn it into a set and looping over it.
* The above thoughts don't seem to jive with setting host from set(reduce([...]) as there will be only one value for host after doing this?